pkg/stockings/fingopack.go

```diff
@@ -0,0 +1,106 @@
+package stockings
+
+import (
+	"errors"
+	"time"
+
+	"github.com/piquette/finance-go"
+	"github.com/piquette/finance-go/chart"
+	"github.com/piquette/finance-go/datetime"
+	"github.com/piquette/finance-go/quote"
+
+	"github.com/shopspring/decimal"
+)
+
+type FingoPack struct {
+}
+
+type piqHistoricalStocks []finance.ChartBar
+
+func (piq FingoPack) GetCurrentPrice(ticker string) (*Stock, error) {
+
+	quote, err := quote.Get(ticker)
+	if err != nil {
+		return nil, err
+	}
+
+	symbol := quote.Symbol
+	name := quote.ShortName
+	currentPrice := decimal.NewFromFloat(quote.RegularMarketPrice)
+	priceChange := float32(quote.RegularMarketChange)
+	percentChange := float32(quote.RegularMarketChangePercent)
+
+	return &Stock{symbol, name, currentPrice, priceChange, percentChange}, nil
+
+}
+
+// we use the GetHistoricalRange function and set our start date to 5 days prior becuase there has to be a valid open market date in the range
+func (piq FingoPack) GetHistoricalPrice(ticker string, date time.Time) (*HistoricalStock, error) {
+
+	api := FingoPack{}
+	historicalQuotes, err := api.GetHistoricalRange(ticker, date.AddDate(0, 0, -5), date)
```

Comment:
bro u just call `piq.GetHistoricalRange` (`piq` from line 38 is basically `self` in python)